frontend/src/scenes/settings/user/TwoFactorAuthentication.tsx

```diff
@@ -1,53 +1,145 @@
+                    <div className="max-w-xl">
                         <b>
                             Use an authenticator app like Google Authenticator or 1Password to scan the QR code below.
                         </b>
                         <Setup2FA
                             onSuccess={() => {
-                                setModalVisible(false)
-                                updateUser({})
-                                loadMemberUpdates()
+                                setSetupModalVisible(false)
+                                handleSuccess()
                             }}
                         />
-                    </>
+                    </div>
+                </LemonModal>
+            )}
+
+            {disableModalVisible && (
+                <LemonModal
+                    title="Disable 2FA"
+                    onClose={() => setDisableModalVisible(false)}
+                    footer={
+                        <>
+                            <LemonButton onClick={() => setDisableModalVisible(false)}>Cancel</LemonButton>
+                            <LemonButton
+                                type="primary"
+                                status="danger"
+                                onClick={() => {
+                                    disable2FA()
+                                    setDisableModalVisible(false)
+                                    handleSuccess()
+                                }}
+                            >
+                                Disable 2FA
+                            </LemonButton>
+                        </>
+                    }
+                >
+                    <p>
+                        Are you sure you want to disable two-factor authentication? This will make your account less
+                        secure.
+                    </p>
+                </LemonModal>
+            )}
+
+            {backupCodesModalVisible && (
+                <LemonModal title="Backup Codes" onClose={() => setBackupCodesModalVisible(false)}>
+                    <div className="space-y-4 max-w-md">
+                        <p>
+                            Save these backup codes in a secure location. Each code can only be used once to sign in if
+                            you lose access to your authentication device.
+                        </p>
+                        {status?.backup_codes?.length ? (
+                            <div className="bg-bg-3000 p-4 rounded font-mono space-y-1 relative">
+                                <LemonButton
+                                    icon={<IconCopy />}
+                                    size="small"
+                                    className="absolute top-4 right-4"
+                                    onClick={() => {
+                                        void navigator.clipboard.writeText(status.backup_codes.join('\n') || '')
```

Comment:
There is a `copyToClipboard` that does this with a fallback and som other niceties that is worth using